interactions/api/gateway.py

```diff
@@ -297,6 +297,14 @@ def check_sub_auto(option) -> tuple:
                         __import__(path),
                         _name,
                     )
+                    if event.lower() in [
+                        "message_create",
+                        "channel_create",
+                        "guild_member_add",
+                        "guild_create",
+                        "guild_role_create",
+                    ]:
+                        data["client"] = self.http
```

Comment:
Why are you building a special condition for the event to allow an `HTTPClient`? A better approach would be doing `if event.lower() in ("_create", "_add")` to check for generic events that allow subsequent actioning.